aesd-char-driver/aesd-circular-buffer.c

```diff
@@ -82,8 +84,35 @@ void aesd_circular_buffer_add_entry(struct aesd_circular_buffer *buffer, const s
     if (buffer->in_offs == buffer->out_offs) {
         buffer->full = true;
     }
+
+    return overwritten;
+}
+
+
+extern void aesd_circular_buffer_free(struct aesd_circular_buffer *buffer) {
+    int entry_num = buffer->out_offs;
+
+    // empty case
+    if ((entry_num == buffer->in_offs) && !(buffer->full)) {
+        return;
+    }
+
+    do {
+#ifdef __KERNEL__
+        kfree(buffer->entry[entry_num].buffptr);
```

Comment:
@vido2373 I'd suggest handling the buffer free in the application rather than in this shared function.  Since the application is providing the buffer it will know whether it ws dynamically allocated.